hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/InternalSchemaUtils.java

```diff
@@ -54,29 +58,75 @@ private InternalSchemaUtils() {
    */
   public static InternalSchema pruneInternalSchema(InternalSchema schema, List<String> names) {
     // do check
-    List<Integer> prunedIds = names.stream().map(name -> {
+    List<Integer> prunedIds = names.stream()
+        .filter(name -> {
+          int id = schema.findIdByName(name);
+          if (id < 0) {
+            LOG.warn(String.format("cannot prune col: %s does not exist in hudi table", name));
```

Comment:
prior to this patch, we were throwing exception and now we are not? is this change intended?